### ISSUE TITLE
Notebook 03. Mistake with confusion matrix  (10 section)?

### ISSUE THREAD

**Issue by AnatolyAdamovich** [mrdbourke/pytorch-deep-learning#163]
Hey,
In the screenshot below we apply `softmax` function with  `dim=0`,  but  `y_logit` has the shape `[32, 10]`. If we apply `accuracy` metric to evaluate score between `y_pred_tensor` and `test_data.targets` we get the lower results than it were before.
![image](https://user-images.githubusercontent.com/61744572/200144161-7fbc29c4-e45f-4ac6-970e-5c6c6e834ae4.png)

Should we use `dim=1`  in `softmax` instead? In this case we get the same results (in terms of accuracy)
Maybe it depends on PyTorch version.

Thank you!

**Comment by ryujin-akm** [comment]
Hi [AnatolyAdamovich](https://github.com/AnatolyAdamovich),

I think it should be `dim = 1`. The reason being `dim = 1`, will account for the probabilities across the `10 columns` ,  `dim = 0`, on the other hand is across the `32 rows`. 
Please correct me if I am wrong.

**Comment by AnatolyAdamovich** [comment]
You are absolutely right, @ryujin-akm

**Comment by mrdbourke** [comment]
Hi @AnatolyAdamovich and @ryujin-akm,

You are both right.

The correct code should be:

```python
# Import tqdm for progress bar
from tqdm.auto import tqdm

# 1. Make predictions with trained model
y_preds = []
model_2.eval()
with torch.inference_mode():
  for X, y in tqdm(test_dataloader, desc="Making predictions"):
    # Send data and targets to target device
    X, y = X.to(device), y.to(device)
    # Do the forward pass
    y_logit = model_2(X)
    # Turn predictions from logits -> prediction probabilities -> predictions labels
    y_pred = torch.softmax(y_logit, dim=1).argmax(dim=1)
    # Put predictions on CPU for evaluation
    y_preds.append(y_pred.cpu())
# Concatenate list of predictions into a tensor
y_pred_tensor = torch.cat(y_preds)
```

Just updated this in the notebook: https://github.com/mrdbourke/pytorch-deep-learning/blob/main/03_pytorch_computer_vision.ipynb

Going to also put a note on the video(s) on ZTM/Udemy.
